source/containers/document-pii-detection/parsers/image_parser.py

```diff
@@ -0,0 +1,79 @@
+
+import os
+from .parser import BaseParser
+
+from PIL import Image
+import numpy as np
+
+from .image_analysis.face_detection import face_detection_main
+from .image_analysis.general_ocr import ocr_main
+
+def check_keywords_exist(det_results, keywords):
+    for keyword in keywords:
+        found = False
+        for dt_result in det_results:
+            text, score = dt_result[1]
+            if keyword in text and score >= 0.5:
+                found = True
+                break
+        if not found:
+            return False
+    return True
+    
+class ImageParser(BaseParser):
+    def __init__(self, s3_client, fd_model_path, ocr_model_path):
+        super().__init__(s3_client=s3_client)
+        self.face_detection_model = face_detection_main.SCRFD(model_file = fd_model_path + 'det.onnx')
+        self.ocr_model = ocr_main.TextSystem(model_path = ocr_model_path)
+        # additional PdfParser constructor code here
+    
+    def read_img(self, file_path):
+        img = np.array(Image.open(file_path).convert('RGB'))[:, :, :3]
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource returned by the following method call: *open*. The resource is allocated by call *Image.open*. Execution paths that do not contain closure statements were detected. To prevent this resource leak, close the object returned by *open()* in a try-finally block or declare it using a `with` statement.

**More info**
[View details about the `with` statement in the Python developer&#39;s guide](https://www.python.org/dev/peps/pep-0343/) (external link).

Reply:
Resolve by using `with Image.open()` to load image.